lib/txRequest/txRequest.js

```diff
@@ -93,10 +95,14 @@ class TxRequest {
     }
 
     if (this.temporalUnit == 1) {
-      return new BigNumber(await Util.getBlockNumber(this.web3))
+      // The reason for the `plus(1)` here is that the next block to be mined
+      // is for all intents and purposes the `now` since the soonest this transaction
+      // could be included is alongside it.
+      return new BigNumber(await Util.getBlockNumber(this.web3)).plus(1);
     } else if (this.temporalUnit == 2) {
-      const timestamp = await Util.getTimestamp(this.web3)
-      return new BigNumber(timestamp)
+      const timestamp = new BigNumber(await Util.getTimestamp(this.web3));
+      const local = new BigNumber(moment().seconds());
```

Comment:
Not sure if using moment for this is worth.

```suggestion
      const local = parseInt(new Date().getTime()/1000);
```

Reply:
Yeah this is fine but I prefer to keep everything as BigNumber instances.